File: 2272. Substring With Largest Variance/Solution.cpp
class Solution {
public:
    int largestVariance(string s) {
        int high = 0;
        string ch = "abcdefghijklmnopqrstuvwxyz";
        for (int i = 0; i < 26; i++) {
            for (int j = i + 1; j < 26; j++) {
                char a = ch[i], b = ch[j];
                int curr = 0, currHigh = 0, currLow = 0, prevHigh = 0, prevLow = 0;
                bool hasA = false, hasB = false;
                for (char c : s) {
                    if (c == a) {
                        hasA = true;
                        curr++;
                        currHigh = max(currHigh, prevHigh);
                        prevHigh = curr;
                    } else if (c == b) {
                        hasB = true;
                        curr--;
                        currLow = min(currLow, prevLow);
                        prevLow = curr;
                    }
                    if (hasA && hasB) high = max(high, max(curr - currLow, currHigh - curr));
                }
            }
        }
        return high;
    }
};

File: 2272. Substring With Largest Variance/Solution.py
class Solution:
    def largestVariance(self, s: str) -> int:
        high = 0
        ch = 'abcdefghijklmnopqrstuvwxyz'
        for i in range(26):
            for j in range(i + 1, 26):
                a, b = ch[i], ch[j]
                curr = 0
                currHigh, currLow = 0, 0
                prevHigh, prevLow = 0, 0
                hasA, hasB = False, False
                for c in s:
                    if c == a:
                        hasA = True
                        curr += 1
                        currHigh = max(prevHigh, currHigh)
                        prevHigh = curr
                    elif c == b:
                        hasB = True
                        curr -= 1
                        currLow = min(prevLow, currLow)
                        prevLow = curr
                    if hasA and hasB:
                        high = max(high, curr - currLow, currHigh - curr)
        return high
